lib/mayaUsd/fileio/translators/translatorMaterial.cpp

```diff
@@ -71,15 +70,22 @@ UsdMayaTranslatorMaterial::Read(
         return shadingEngine;
     }
 
-    if (UsdMayaShadingModeImporter importer =
-            UsdMayaShadingModeRegistry::GetImporter(jobArguments.shadingMode)) {
-        shadingEngine = importer(&c, jobArguments);
-    }
+    UsdMayaJobImportArgs localArguments = jobArguments;
+    for (const auto& shadingMode : jobArguments.shadingModes) {
+        if (shadingMode.first == UsdMayaShadingModeTokens->none) {
+            break;
+        }
+        if (UsdMayaShadingModeImporter importer
+            = UsdMayaShadingModeRegistry::GetImporter(shadingMode.first)) {
+            localArguments.shadingModes = UsdMayaJobImportArgs::ShadingModes(1, shadingMode);
+            shadingEngine = importer(&c, localArguments);
+        }
 
-    if (!shadingEngine.isNull()) {
-        c.AddCreatedObject(shadeMaterial.GetPrim(), shadingEngine);
+        if (!shadingEngine.isNull()) {
+            c.AddCreatedObject(shadeMaterial.GetPrim(), shadingEngine);
+            return shadingEngine;
```

Comment:
Stop searching when an importer succeeds in crating a shading engine.
Does not work with importers that could share existing shading engines by adding specialized surface ports (RenderMan  is a known example). To be added later.